monkey/monkey_island/cc/ui/src/components/map/MapPageWrapper.tsx

```diff
@@ -34,18 +34,18 @@ const MapPageWrapper = (props) => {
   const [graphSnapshot, setGraphSnapshot] = useState<Graph>({edges: [], nodes: []});
 
 
-  async function fetchMapNodes() {
-    await getCollectionObject(APIEndpoint.nodes, 'machine_id').then(nodeObj => setNodes(nodeObj));
-    await getCollectionObject(APIEndpoint.machines, 'id').then(machineObj => setMachines(machineObj));
-    await getAllAgents().then(agents => setAgents(agents.sort()));
-    return getPropagationEvents().then(events => setPropagationEvents(events));
+  async function fetchMapNodes(refreshToken: boolean) {
+    await getCollectionObject(APIEndpoint.nodes, 'machine_id', refreshToken).then(nodeObj => setNodes(nodeObj));
+    await getCollectionObject(APIEndpoint.machines, 'id', refreshToken).then(machineObj => setMachines(machineObj));
+    await getAllAgents(refreshToken).then(agents => setAgents(agents.sort()));
+    return getPropagationEvents(refreshToken).then(events => setPropagationEvents(events));
   }
 
   useEffect(() => {
     if (!updateInProgress) {
       let oneSecond = 1000;
       setUpdateInProgress(true);
-      fetchMapNodes()
+      fetchMapNodes(updateInProgress)
```

Comment:
My number one criteria for this page is that if you walk away from it, it should time out. This shouldn't be dependent on what agents are doing.